File: next/kmp/core/src/commonMain/kotlin/org/dweb_browser/core/ipc/helper/EndpointIpcMessage.kt
package org.dweb_browser.core.ipc.helper

import kotlinx.serialization.Serializable
import org.dweb_browser.helper.OrderBy


/**分发消息到各个ipc的监听时使用
 *
 * 这里的 orderBy 和 IpcMessage 里头的 orderBy 不冲突，是两层各自去排序
 */
@Serializable
data class EndpointIpcMessage(
  val pid: Int,
  val ipcMessage: IpcMessage,
) : EndpointMessage(ENDPOINT_MESSAGE_TYPE.IPC)

File: desktop-dev/src/core/ipc/endpoint/EndpointIpcMessage.ts
import { $IpcMessage } from "../helper/const.ts";
import { ENDPOINT_MESSAGE_TYPE } from "./EndpointMessage.ts";
import { endpointMessageBase } from "./internal/EndpointMessage.ts";

export type $EndpointIpcMessage<M extends $IpcMessage> = ReturnType<typeof endpointIpcMessage<M>>;

export const endpointIpcMessage = <M extends $IpcMessage>(pid: number, ipcMessage: M) =>
  ({
    ...endpointMessageBase(ENDPOINT_MESSAGE_TYPE.IPC),
    pid,
    ipcMessage,
  } as const);
